hyperprocess_macro/src/lib.rs

```diff
@@ -766,9 +945,20 @@ fn generate_async_handler_arm(
         // Updated pattern to match struct variant with no fields
         quote! {
             HPMRequest::#variant_name{} => {
+                // Capture context values before async execution
+                let current_path = hyperware_app_common::get_path();
+                let current_method = hyperware_app_common::get_http_method();
+                
                 // Create a raw pointer to state for use in the async block
                 let state_ptr: *mut #self_ty = state;
                 hyperware_app_common::hyper! {
+                    // Restore context in the async task
+                    hyperware_app_common::APP_HELPERS.with(|ctx| {
+                        let mut ctx_mut = ctx.borrow_mut();
+                        ctx_mut.current_path = current_path;
+                        ctx_mut.current_http_method = current_method;
+                    });
```

Comment:
is capturing and restoring context like this required? See https://github.com/hyperware-ai/hyperprocess-macro/pull/19/commits/3568d35b461ce4d54075a9b5c02cd521f5527efb which seems to work fine without. If we need it, please add a test that shows how removing this breaks

similar for below